File: app/src/main/java/com/example/mindpulse/ui/profile/ProfileFragment.java
package com.example.mindpulse.ui.profile;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;

import com.example.mindpulse.R;
import com.example.mindpulse.databinding.FragmentProfileBinding;
import com.example.mindpulse.ui.auth.AuthViewModel;

public class ProfileFragment extends Fragment {

    private FragmentProfileBinding binding;
    private AuthViewModel authViewModel;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentProfileBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        authViewModel = new ViewModelProvider(requireActivity()).get(AuthViewModel.class);

        setupUI();
        observeViewModel();
    }

    private void setupUI() {
        binding.btnLogout.setOnClickListener(v -> {
            authViewModel.logout();
        });

        binding.btnEditProfile.setOnClickListener(v -> {
            // Placeholder for edit profile
        });

        binding.btnSettings.setOnClickListener(v -> {
            Navigation.findNavController(requireView()).navigate(R.id.action_profile_to_settings);
        });
    }

    private void observeViewModel() {
        authViewModel.getUserLiveData().observe(getViewLifecycleOwner(), firebaseUser -> {
            if (firebaseUser == null) {
                Navigation.findNavController(requireView()).navigate(R.id.action_login_to_home); // Assuming global action or specific logout handling
                // Actually, since this is ProfileFragment, we should navigate to Login
                Navigation.findNavController(requireView()).navigate(R.id.navigation_login);
            } else {
                binding.tvEmail.setText(firebaseUser.getEmail());
                // Set name if available in user object/Firestore
            }
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/mindpulse/ui/profile/ProfileFragment.kt
package com.example.mindpulse.ui.profile

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.navigation.fragment.findNavController
import com.example.mindpulse.R
import com.example.mindpulse.databinding.FragmentProfileBinding
import com.example.mindpulse.ui.auth.AuthViewModel
import com.example.mindpulse.ui.auth.AuthViewModelFactory

class ProfileFragment : Fragment() {

    private var _binding: FragmentProfileBinding? = null
    private val binding get() = _binding!!

    private val authViewModel: AuthViewModel by activityViewModels {
        AuthViewModelFactory()
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentProfileBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setupUI()
        observeViewModel()
    }

    private fun setupUI() {
        binding.btnLogout.setOnClickListener {
            authViewModel.logout()
        }

        binding.btnEditProfile.setOnClickListener {
            // Placeholder for edit profile
        }

        binding.btnSettings.setOnClickListener {
            findNavController().navigate(R.id.action_profile_to_settings)
        }

        binding.btnEmergencyContacts.setOnClickListener {
            findNavController().navigate(R.id.navigation_emergency)
        }
    }

    private fun observeViewModel() {
        authViewModel.currentUser.observe(viewLifecycleOwner) { user ->
            if (user == null) {
                findNavController().navigate(R.id.navigation_login)
            } else {
                binding.tvName.text = user.fullName.ifEmpty { "User Name" }
                binding.tvEmail.text = user.email
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
